docs/changelog.md

```diff
@@ -16,6 +16,15 @@ v0.27 + 1
 * You can now swap out memory allocators via the
   `GIT_OPT_SET_ALLOCATOR` option with `git_libgit2_opts()`.
 
+* You can now ensure that functions do not discard unwritten changes to the
+  index via the `GIT_OPT_ENABLE_UNSAVED_INDEX_SAFETY` option to
+  `git_libgit2_opts()`.  This will cause functions that implicitly re-read
+  the index (eg, `git_checkout`) to fail if you have staged changes to the
+  index but you have not written the index to disk.
+
+  At present, this defaults to off, but we intend to enable this more
+  broadly in the future, as a warning or error.
```

Comment:
Should we urge consumers to review how they handle dirty indices and then enable the option?